src/main/java/com/synopsys/integration/alert/web/api/job/JobConfigActions.java

```diff
@@ -56,150 +62,183 @@
+                Collection<ConfigurationFieldModel> savedFieldsModels = modelConverter.convertToConfigurationFieldModelMap(beforeSaveEventFieldModel).values();
+                configurationFieldModels.addAll(savedFieldsModels);
+            }
+            ConfigurationJobModel savedJob = configurationAccessor.createJob(descriptorNames, configurationFieldModels);
+            JobFieldModel savedJobFieldModel = convertToJobFieldModel(savedJob);
 
-        Set<FieldModel> updatedFieldModels = new HashSet<>();
-        for (FieldModel fieldModel : savedJobFieldModel.getFieldModels()) {
-            FieldModel updatedModel = fieldModelProcessor.performAfterSaveAction(fieldModel);
-            updatedFieldModels.add(updatedModel);
+            Set<FieldModel> updatedFieldModels = new HashSet<>();
+            for (FieldModel fieldModel : savedJobFieldModel.getFieldModels()) {
+                FieldModel updatedModel = fieldModelProcessor.performAfterSaveAction(fieldModel);
+                updatedFieldModels.add(updatedModel);
+            }
+            savedJobFieldModel.setFieldModels(updatedFieldModels);
+            return new ActionResponse<>(HttpStatus.OK, savedJobFieldModel);
+        } catch (AlertException ex) {
+            logger.error("Error creating job", ex);
+            return new ActionResponse<>(HttpStatus.INTERNAL_SERVER_ERROR, ex.getMessage());
         }
-        savedJobFieldModel.setFieldModels(updatedFieldModels);
-        return savedJobFieldModel;
     }
 
-    public JobFieldModel updateJob(UUID id, JobFieldModel jobFieldModel) throws AlertException {
-        MessageResult validationResult = validateJob(jobFieldModel);
-        validationResult.throwExceptionForFieldStatues();
-        validateJobNameUnique(id, jobFieldModel);
-
-        ConfigurationJobModel previousJob = configurationAccessor.getJobById(id)
-                                                .orElseThrow(() -> new IllegalStateException("No previous job present when the only possible valid state for this stage of the method would require it"));
-        Map<String, FieldModel> descriptorAndContextToPreviousFieldModel = new HashMap<>();
-        for (ConfigurationModel previousJobConfiguration : previousJob.getCopyOfConfigurations()) {
-            FieldModel previousJobFieldModel = modelConverter.convertToFieldModel(previousJobConfiguration);
-            descriptorAndContextToPreviousFieldModel.put(previousJobFieldModel.getDescriptorName() + previousJobFieldModel.getContext(), previousJobFieldModel);
-        }
+    @Override
+    protected ActionResponse<JobFieldModel> updateAfterChecks(UUID id, JobFieldModel resource) {
+        try {
+            ConfigurationJobModel previousJob = configurationAccessor.getJobById(id)
+                                                    .orElseThrow(() -> new IllegalStateException("No previous job present when the only possible valid state for this stage of the method would require it"));
```

Comment:
I think this should actually be a 404 case.

Reply:
in the update method a 404 not found is already returned. If we are executing here and cannot find it it's an issue.